diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 74ecc8bba..f68562c9f 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -199,7 +206,7 @@ set(PL_LIB_INDEX ${SWIPL_BUILD_LIBRARY}/INDEX.pl)
 # FIXME: we should create these in the build directory
 add_custom_target(
     vmi-metadata
-    COMMAND mkvmi ${CMAKE_CURRENT_SOURCE_DIR}
+    COMMAND ./mkvmi-host ${CMAKE_CURRENT_SOURCE_DIR}
     BYPRODUCTS pl-vmi.h pl-codetable.ic pl-jumptable.ic
     DEPENDS pl-vmi.c
     COMMENT "Generating VMI metadata"
@@ -207,7 +214,7 @@ add_custom_target(
 
 add_custom_target(
     core-constants
-    COMMAND defatom ${CMAKE_CURRENT_SOURCE_DIR}
+    COMMAND ./defatom-host ${CMAKE_CURRENT_SOURCE_DIR}
     BYPRODUCTS pl-atom.ih pl-funct.ih
     DEPENDS ATOMS
     COMMENT "Generating static atoms and functors"
