diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 74ecc8bba..e899de733 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -229,7 +229,7 @@ add_custom_command(
 add_custom_command(
     OUTPUT  ${PL_LIB_INDEX}
     COMMAND swipl -f none --nopce -q
-            -g "\"make_library_index(swi(library))\"" -t halt
+            -g "\"make_library_index('${CMAKE_BINARY_DIR}/home/library')\"" -t halt
     DEPENDS ${SWIPL_BOOT_FILE} prolog_home
 )
 
