diff --git a/packages/jpl/CMakeLists.txt b/packages/jpl/CMakeLists.txt
index 1eeaa56..57081ef 100644
--- a/packages/jpl/CMakeLists.txt
+++ b/packages/jpl/CMakeLists.txt
@@ -39,9 +39,13 @@ if(NOT JAVA_HOME)
   endif()
 endif()
 
-find_package(JNI)
+if(NOT ANDROID)
+   find_package(JNI)
+endif(NOT ANDROID)
+
 find_package(Java COMPONENTS Development)
-if(JNI_FOUND AND Java_FOUND)
+if( (NOT ANDROID AND (JNI_FOUND AND Java_FOUND))
+     OR (ANDROID AND Java_FOUND) )
 include(JUnit)
 
 add_subdirectory(src/java)
@@ -138,4 +142,4 @@ pkg_doc(
 	SOURCE jpl.pl jpldoc.tex
     DEPENDS jpl)
 
-endif(JNI_FOUND AND Java_FOUND)
+endif()
