diff --git a/packages/jpl/src/c/jpl.c b/packages/jpl/src/c/jpl.c
index 050cf56..d0cae7b 100644
--- a/packages/jpl/src/c/jpl.c
+++ b/packages/jpl/src/c/jpl.c
@@ -63,7 +63,7 @@ refactoring (trivial):
 
 /*#define DEBUG(n, g) ((void)0) */
 #ifndef DEBUG_LEVEL
-#define DEBUG_LEVEL 3
+#define DEBUG_LEVEL 0
 #endif
 #define DEBUG(n, g) (n >= DEBUG_LEVEL ? g : (void)0)
 
@@ -1836,8 +1836,9 @@ jni_create_jvm_c(char *classpath)
     strcpy(cpoptp, "-Djava.class.path="); /* was cpopt */
     strcat(cpoptp, classpath);            /* oughta check length... */
     vm_args.options        = opt;
-    opt[optn].optionString = cpoptp; /* was cpopt */
-    optn++;
+    opt[optn++].optionString = cpoptp; /* was cpopt */
+    opt[optn++].optionString = "-verbose:jni";
+    //optn++;
   }
   /* opt[optn++].optionString = "-Djava.compiler=NONE"; */
   /* opt[optn].optionString = "exit";       // I don't understand this yet... */
@@ -1872,6 +1873,7 @@ jni_create_jvm_c(char *classpath)
   vm_args.nOptions = optn;
   /* vm_args.ignoreUnrecognized = TRUE; */
 
+  DEBUG(1, Sdprintf("[creating JVM after thread_stack]\n"));
   return (JNI_GetCreatedJavaVMs(&jvm, 1, &n) ==
                       0 /* what does the '1' arg mean? */
                   && n == 1
