Submodule packages/ssl contains modified content
diff --git a/packages/ssl/CMakeLists.txt b/packages/ssl/CMakeLists.txt
index 698a8ea..0ec9acc 100644
--- a/packages/ssl/CMakeLists.txt
+++ b/packages/ssl/CMakeLists.txt
@@ -110,12 +110,19 @@ if(NOT DEFINED SYSTEM_CACERT_FILENAME)
       "Location of the system certificate file")
 endif()
 
-find_program(OPENSSL openssl${CMAKE_EXECUTABLE_SUFFIX}
-	     HINTS ${OPENSSL_ROOT_DIR}/bin)
+#For cross-compiling with need to use the host's binary
+#so we first search in the PATH environment variable
+#if not found then it tries all the cmake defualt paths
+#along with OPENSSL_ROOT_DIR/bin
+find_program(OPENSSLPRG openssl${CMAKE_EXECUTABLE_SUFFIX}
+   PATHS /usr/bin /bin /usr/sbin /sbin /usr/local/bin /usr/local/sbin
+   NO_DEFAULT_PATH)
+#find_program(OPENSSLPRG openssl${CMAKE_EXECUTABLE_SUFFIX}
+#	     HINTS ${OPENSSL_ROOT_DIR}/bin)
 
 configure_file(config.h.cmake config.h)
 
-if(OPENSSL AND EXISTS "${OPENSSL}")
+if(OPENSSLPRG AND NOT EXISTS "${OPENSSLPRG}")
   configure_file(mkcerts.pl.in mkcerts.pl @ONLY)
   set(SSL_TESTS ON)
 else()
diff --git a/packages/ssl/mkcerts.pl.in b/packages/ssl/mkcerts.pl.in
index 48fb288..9d953f2 100644
--- a/packages/ssl/mkcerts.pl.in
+++ b/packages/ssl/mkcerts.pl.in
@@ -50,7 +50,8 @@
 
 :- initialization(main, main).
 
-openssl_executable('@OPENSSL@').
+%openssl_executable('@OPENSSLPRG@').
+openssl_executable('/usr/bin/openssl').
 
 /** <module> Create SSL test certificates
 
